### PR TITLE
K8 create

### PR DESCRIPTION
**Pull request by wporr**
This will allow create to be deployed on kubernetes. Now, instead of pinning to dataloss, our worker mounts all necessary directories over NFS. I also added an argument to the healthcheck script so that you can specify a custom configuration file.

### CODE REVIEW

File: kubernetes/k8_create.yml.erb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      app: create
+  template:
+      metadata:
+        labels:
+          app: create
+      spec:
+        containers:
+          - name: create-worker
+            image: "docker.ocf.berkeley.edu/create:<%= version %>"
+            imagePullPolicy: Always
```
Comment (kkuehlz) [inline kubernetes/k8_create.yml.erb]: What is the reasoning for this?

**Review by kkuehlz — APPROVED** [review]
Quality code.

File: kubernetes/k8_create.yml.erb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                name: create-secrets
+                readOnly: true
+            livenessProbe:
+              exec:
+                command:
+                - python
+                - '-m'
+                - 'create.healthcheck'
+              initialDelaySeconds: 15
+              periodSeconds: 15
```
Comment (cg505) [inline kubernetes/k8_create.yml.erb]: maybe we should define an equivalent readiness probe

File: kubernetes/k8_create.yml.erb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+              - containerPort: 6378
+            resources:
+              limits:
+                cpu: 750m
+                memory: 512Mi
+            volumeMounts:
+              - mountPath: /home
+                name: nfs-export-home
+              - mountPath: /services
+                name: nfs-export-services
```
Comment (cg505) [inline kubernetes/k8_create.yml.erb]: mentioned it in the puppet PR, but is this actually needed?

File: kubernetes/k8_create.yml.erb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      metadata:
+        labels:
+          app: create
+      spec:
+        containers:
+          - name: create-worker
+            image: "docker.ocf.berkeley.edu/create:<%= version %>"
+            imagePullPolicy: Always
+            ports:
+              - containerPort: 6378
```
Comment (cg505) [inline kubernetes/k8_create.yml.erb]: Why is this needed? Is this the port Celery listens on? If so, don't we need a k8s `service` as well?

**Review by cg505 — COMMENTED** [review]
I'd rename the yaml file to `create-deployment.yml.erb` to keep more in line with how we usually name these files. (It's also just a more descriptive name.)

You should also update the `Jenkinsfile` and remove the `deployTargets`.

File: kubernetes/k8_create.yml.erb (hunk 1 above)
Comment (wporr) [reply kubernetes/k8_create.yml.erb]: Kubernetes does not always pull the latest image from the docker repo, so if we update the image and then redeploy the service, kube will keep the old image instead of re-pulling. Not sure how necessary this is for production, but it was painful for debugging.

File: kubernetes/k8_create.yml.erb (hunk 3 above)
Comment (wporr) [reply kubernetes/k8_create.yml.erb]: Yes, see comment in other pr

File: kubernetes/k8_create.yml.erb (hunk 2 above)
Comment (wporr) [reply kubernetes/k8_create.yml.erb]: From what I understand, a readiness probe would only signify when the pods are ready to receive traffic, but since celery only interacts through redis queues that wouldn't be necessary. So I left it out. If theres another reason for having it ill put it in

File: kubernetes/k8_create.yml.erb (hunk 4 above)
Comment (wporr) [reply kubernetes/k8_create.yml.erb]: Not strictly necessary, mostly just informational.

File: kubernetes/k8_create.yml.erb (hunk 1 above)
Comment (kkuehlz) [reply kubernetes/k8_create.yml.erb]: All of the images use a version which Jenkins will bump, so I don't think this is necessary.

File: kubernetes/k8_create.yml.erb (hunk 2 above)
Comment (cg505) [reply kubernetes/k8_create.yml.erb]: when a new build is deployed, kube won't bring down the old containers until the new ones are ready

Comment (wporr) [reply kubernetes/k8_create.yml.erb]: Alright then I'll add it

File: kubernetes/k8_create.yml.erb (hunk 4 above)
Comment (cg505) [reply kubernetes/k8_create.yml.erb]: What is it though?

File: kubernetes/k8_create.yml.erb (hunk 1 above)
Comment (cg505) [reply kubernetes/k8_create.yml.erb]: In production all the versions will be distinct so the same tag will never update

**Comment by wporr** [comment]
If theres no more changes I'll squash and merge
